test/tests/functional/pbs_multi_sched.py

```diff
@@ -1322,3 +1322,50 @@ def test_list_multi_sched(self):
             err_msg = "Unknown Scheduler"
             self.assertTrue(err_msg in e.msg[0],
                             "Error message is not expected")
+
+    def test_multisched_sched_cycle_length(self):
+        """
+        Test the scheduler attribute sched_cycle_length for multisched
+        """
+        self.server.manager(MGR_CMD_SET, SERVER, {'scheduling': 'False'},
+                            expect=True)
+        self.server.manager(MGR_CMD_SET, SCHED,
+                            {'sched_cycle_length': '00:00:01'},
+                            id='default',
+                            expect=True)
+        a = {'resources_available.ncpus': 100}
```

Comment:
I feel that relying on number of jobs to ensure that scheduler will take more than a second to run is not the right approach. 
If you really want to delay the scheduling cycle then I strongly suggest you add server dynamic resource script and sleep in that script before printing a resource value. This will ensure that scheduler exceeds the cycle time in all cases (slow or fast machines).

Reply:
@arungrover I have tried to do the same as you suggested but observed that the server dynamic resource blocks the sched_cycle_length till the dynamic resource script finishes it's execution.

Reply:
Scheduler getting blocked until the time server_dyn_res script completes execution is not a bug. This is as per the current design. According to my knowledge there are some enhancements/improvements to server_dyn_res like introducing a timeout which can be configurable etc. 

It seems scheduling cycle start time is calculated in the function man_sched_loop. But server_dyn_res script is executed much before this function i.e. query_server itself where we are not storing the cycle start time. Due to this you cannot see the message desired in the scenario that is tried.

May be the initial thought could be to calculate the cycle start time when scheduler truly starts scheduling jobs i.e. after all its query of server, execution of server dynamic resources etc.

Reply:
@Bhagat-Rajput , In the interest of time for the time being you can actually work on the other test case as discussed.  The current test case once we have more info can always be added to Multisched test suite later also.

Reply:
@suresh-thelkar I have added the job_sort_formula_threshold test case as you suggested.